src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -1410,7 +1410,7 @@ def forward(
                     f"unexpected if using padding tokens in conjunction with `inputs_embeds.`"
                 )
 
-        pooled_logits = logits[range(batch_size), sequence_lengths]
+        pooled_logits = logits[torch.arange(batch_size), sequence_lengths]
```

Comment:
```suggestion
        pooled_logits = logits[torch.arange(batch_size, device=self.device), sequence_lengths]
```
We need to make sure the tensor is on the same device no?